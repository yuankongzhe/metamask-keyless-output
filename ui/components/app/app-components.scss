/** Please import your files in alphabetical order **/
@import 'account-list-item/index';
@import 'add-network/index';
@import 'app-loading-spinner/index';
@import 'alerts/alerts';
@import 'beta-header/index';
@import 'cancel-speedup-popover/index';
@import 'import-token/index';
@import 'nfts-items/index';
@import 'nfts-tab/index';
@import 'nft-details/index';
@import 'nft-default-image/index';
@import 'nft-options/index';
@import 'connected-accounts-list/index';
@import 'connected-accounts-permissions/index';
@import 'connected-sites-list/index';
@import 'create-new-vault/create-new-vault.scss';
@import 'flask/experimental-area/index';
@import 'snaps/snap-content-footer/index';
@import 'snaps/snap-install-warning/index';
@import 'snaps/snap-ui-renderer/index';
@import 'snaps/snap-ui-markdown/index';
@import 'snaps/snap-delineator/index';
@import 'snaps/snap-list-item/index';
@import 'snaps/copyable/index';
@import 'snaps/snap-version/index';
@import 'snaps/show-more/index';
@import 'snaps/insight-warnings/index';
@import 'snaps/snap-authorship-header/index';
@import 'hold-to-reveal-button/index';
@import 'home-notification/index';
@import 'info-box/index';
@import 'modal/index';
@import 'modals/index';
@import 'multiple-notifications/index';
@import 'network-display/index';
@import 'incoming-trasaction-toggle/network-toggle';
@import 'permission-page-container/index';
@import 'permissions-connect-footer/index';
@import 'permissions-connect-header/index';
@import 'permissions-connect-permission-list/index';
@import 'permission-cell/index';
@import 'recovery-phrase-reminder/index';
@import 'step-progress-bar/index.scss';
@import 'selected-account/index';
@import 'srp-input/srp-input';
@import 'snaps/snap-privacy-warning/index';
@import 'tab-bar/index';
@import 'token-cell/token-cell';
@import 'token-list-display/token-list-display';
@import 'transaction-activity-log/index';
@import 'transaction-breakdown/index';
@import 'transaction-icon/transaction-icon';
@import 'transaction-list-item-details/index';
@import 'transaction-list-item/index';
@import 'transaction-list/index';
@import 'transaction-status-label/index';
@import 'wallet-overview/index';
@import 'whats-new-popup/index';
@import 'terms-of-use-popup/index';
@import 'loading-network-screen/index';
@import 'currency-input/index';
@import 'detected-token/detected-token-aggregators/index';
@import 'detected-token/detected-token-values/index';
@import 'detected-token/detected-token-details/index';
@import 'detected-token/detected-token-selection-popover/index';
@import 'network-account-balance-header/index';
@import 'name/index';
@import 'name/name-details/index';
///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
@import '../institutional/interactive-replacement-token-notification/index';
@import '../institutional/confirm-remove-jwt-modal/index';
@import '../institutional/custody-confirm-link-modal/index';
@import '../institutional/transaction-failed-modal/index';
///: END:ONLY_INCLUDE_IF
///: BEGIN:ONLY_INCLUDE_IF(snaps)
@import 'snaps/snap-insight/index';
///: END:ONLY_INCLUDE_IF
