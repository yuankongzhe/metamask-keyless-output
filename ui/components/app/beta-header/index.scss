.beta-header {
  &__message {
    text-align: center;
    flex-grow: 1;
  }

  &__button {
    background: transparent;
    padding: 0 6px;
    margin: 0;

    i {
      color: var(--color-warning-inverse);
    }
  }
}
