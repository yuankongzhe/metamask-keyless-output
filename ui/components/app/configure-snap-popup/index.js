export { default, ConfigureSnapPopupType } from './configure-snap-popup';
