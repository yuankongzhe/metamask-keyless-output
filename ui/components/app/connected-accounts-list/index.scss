@use "design-system";

.connected-accounts-list {
  display: flex;
  flex-direction: column;
  align-items: center;

  &__row {
    width: 100%;
    word-break: break-word;
  }

  &__row-content {
    display: flex;
    flex-direction: row;
    align-items: center;
  }
}

.connected-accounts-options {
  &__button {
    font-size: design-system.$font-size-h4;
    background: inherit;
    color: var(--color-icon-default);
  }
}

.tippy-tooltip.none-theme {
  background: none;
  padding: 0;
}
