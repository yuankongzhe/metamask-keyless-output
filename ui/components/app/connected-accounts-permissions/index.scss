.connected-accounts-permissions {
  &__list {
    padding-block: 8px;
    height: 100%;
    overflow: auto;
  }

  &__list-item {
    display: flex;
  }

  &__list-container {
    max-height: 0;
    height: auto;
    transition: max-height 0.8s cubic-bezier(0.4, 0, 0.2, 1);

    &--expanded {
      // arbitrarily set hard coded value for effect to work
      max-height: 100px;
    }
  }
}
