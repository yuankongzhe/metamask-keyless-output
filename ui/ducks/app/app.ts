import { AnyAction, Action } from 'redux';
import { PayloadAction } from '@reduxjs/toolkit';
import {
  WebHIDConnectedStatuses,
  HardwareTransportStates,
} from '../../../shared/constants/hardware-wallets';
import * as actionConstants from '../../store/actionConstants';

interface AppState {
  shouldClose: boolean;
  menuOpen: boolean;
  modal: {
    open: boolean;
    modalState: {
      name: string | null;
      props: Record<string, any>;
    };
    previousModalState: {
      name: string | null;
    };
  };
  alertOpen: boolean;
  alertMessage: string | null;
  qrCodeData: {
    type?: string | null;
    values?: { address?: string | null };
  } | null;
  networkDropdownOpen: boolean;
  importNftsModal: {
    open: boolean;
    tokenAddress?: string;
    tokenId?: string;
    ignoreErc20Token?: boolean;
  };
  showIpfsModalOpen: boolean;
  keyringRemovalSnapModal: {
    snapName: string;
    result: 'success' | 'failure' | 'none';
  };
  showKeyringRemovalSnapModal: boolean;
  importTokensModalOpen: boolean;
  deprecatedNetworkModalOpen: boolean;
  accountDetail: {
    subview?: string;
    accountExport?: string;
    privateKey?: string;
  };
  isLoading: boolean;
  loadingMessage: string | null;
  scrollToBottom: boolean;
  warning: string | null | undefined;
  buyView: Record<string, any>;
  defaultHdPaths: {
    trezor: string;
    ledger: string;
    lattice: string;
  };
  networksTabSelectedRpcUrl: string | null;
  requestAccountTabs: Record<string, number>; // [url.origin]: tab.id
  openMetaMaskTabs: Record<string, boolean>; // openMetamaskTabsIDs[tab.id]): true/false
  currentWindowTab: Record<string, any>; // tabs.tab https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab
  showWhatsNewPopup: boolean;
  showTermsOfUsePopup: boolean;
  singleExceptions: {
    testKey: string | null;
  };
  gasLoadingAnimationIsShowing: boolean;
  smartTransactionsError: string | null;
  smartTransactionsErrorMessageDismissed: boolean;
  ledgerWebHidConnectedStatus: WebHIDConnectedStatuses;
  ledgerTransportStatus: HardwareTransportStates;
  newNftAddedMessage: string;
  removeNftMessage: string;
  newNetworkAddedName: string;
  newNetworkAddedConfigurationId: string;
  selectedNetworkConfigurationId: string;
  sendInputCurrencySwitched: boolean;
  newTokensImported: string;
  newTokensImportedError: string;
  onboardedInThisUISession: boolean;
  customTokenAmount: string;
  txId: string | null;
  accountDetailsAddress: string;
  ///: BEGIN:ONLY_INCLUDE_IF(snaps)
  snapsInstallPrivacyWarningShown: boolean;
  ///: END:ONLY_INCLUDE_IF
}

interface AppSliceState {
  appState: AppState;
}

// default state
const initialState: AppState = {
  shouldClose: false,
  menuOpen: false,
  modal: {
    open: false,
    modalState: {
      name: null,
      props: {},
    },
    previousModalState: {
      name: null,
    },
  },
  alertOpen: false,
  alertMessage: null,
  qrCodeData: null,
  networkDropdownOpen: false,
  importNftsModal: { open: false },
  showIpfsModalOpen: false,
  keyringRemovalSnapModal: {
    snapName: '',
    result: 'none',
  },
  showKeyringRemovalSnapModal: false,
  importTokensModalOpen: false,
  deprecatedNetworkModalOpen: false,
  accountDetail: {
    privateKey: '',
  },
  // Used to display loading indicator
  isLoading: false,
  loadingMessage: null,
  // Used to display error text
  warning: null,
  buyView: {},
  defaultHdPaths: {
    trezor: `m/44'/60'/0'/0`,
    ledger: `m/44'/60'/0'/0/0`,
    lattice: `m/44'/60'/0'/0`,
  },
  networksTabSelectedRpcUrl: '',
  requestAccountTabs: {},
  openMetaMaskTabs: {},
  currentWindowTab: {},
  showWhatsNewPopup: true,
  showTermsOfUsePopup: true,
  singleExceptions: {
    testKey: null,
  },
  gasLoadingAnimationIsShowing: false,
  smartTransactionsError: null,
  smartTransactionsErrorMessageDismissed: false,
  ledgerWebHidConnectedStatus: WebHIDConnectedStatuses.unknown,
  ledgerTransportStatus: HardwareTransportStates.none,
  newNftAddedMessage: '',
  removeNftMessage: '',
  newNetworkAddedName: '',
  newNetworkAddedConfigurationId: '',
  selectedNetworkConfigurationId: '',
  sendInputCurrencySwitched: false,
  newTokensImported: '',
  newTokensImportedError: '',
  onboardedInThisUISession: false,
  customTokenAmount: '',
  scrollToBottom: true,
  txId: null,
  accountDetailsAddress: '',
  ///: BEGIN:ONLY_INCLUDE_IF(snaps)
  snapsInstallPrivacyWarningShown: false,
  ///: END:ONLY_INCLUDE_IF
};

export default function reduceApp(
  state: AppState,
  action: AnyAction,
): AppState {
  const appState: AppState = {
    ...initialState,
    ...state,
  };

  switch (action.type) {
    // dropdown methods
    case actionConstants.NETWORK_DROPDOWN_OPEN:
      return {
        ...appState,
        networkDropdownOpen: true,
      };

    case actionConstants.NETWORK_DROPDOWN_CLOSE:
      return {
        ...appState,
        networkDropdownOpen: false,
      };

    case actionConstants.IMPORT_NFTS_MODAL_OPEN:
      return {
        ...appState,
        importNftsModal: {
          open: true,
          ...action.payload,
        },
      };

    case actionConstants.IMPORT_NFTS_MODAL_CLOSE:
      return {
        ...appState,
        importNftsModal: {
          open: false,
        },
      };

    case actionConstants.SHOW_IPFS_MODAL_OPEN:
      return {
        ...appState,
        showIpfsModalOpen: true,
      };

    case actionConstants.SHOW_IPFS_MODAL_CLOSE:
      return {
        ...appState,
        showIpfsModalOpen: false,
      };

    case actionConstants.IMPORT_TOKENS_POPOVER_OPEN:
      return {
        ...appState,
        importTokensModalOpen: true,
      };

    case actionConstants.IMPORT_TOKENS_POPOVER_CLOSE:
      return {
        ...appState,
        importTokensModalOpen: false,
      };

    case actionConstants.DEPRECATED_NETWORK_POPOVER_OPEN:
      return {
        ...appState,
        deprecatedNetworkModalOpen: true,
      };

    case actionConstants.DEPRECATED_NETWORK_POPOVER_CLOSE:
      return {
        ...appState,
        deprecatedNetworkModalOpen: false,
      };

    // alert methods
    case actionConstants.ALERT_OPEN:
      return {
        ...appState,
        alertOpen: true,
        alertMessage: action.payload,
      };

    case actionConstants.ALERT_CLOSE:
      return {
        ...appState,
        alertOpen: false,
        alertMessage: null,
      };

    case actionConstants.SET_ACCOUNT_DETAILS_ADDRESS: {
      return {
        ...appState,
        accountDetailsAddress: action.payload,
      };
    }

    // qr scanner methods
    case actionConstants.QR_CODE_DETECTED:
      return {
        ...appState,
        qrCodeData: action.value,
      };

    // Smart Transactions errors.
    case actionConstants.SET_SMART_TRANSACTIONS_ERROR:
      return {
        ...appState,
        smartTransactionsError: action.payload,
      };

    case actionConstants.DISMISS_SMART_TRANSACTIONS_ERROR_MESSAGE:
      return {
        ...appState,
        smartTransactionsErrorMessageDismissed: true,
      };

    // modal methods:
    case actionConstants.MODAL_OPEN: {
      const { name, ...modalProps } = action.payload;

      return {
        ...appState,
        modal: {
          open: true,
          modalState: {
            name,
            props: { ...modalProps },
          },
          previousModalState: { ...appState.modal.modalState },
        },
      };
    }

    case actionConstants.MODAL_CLOSE:
      return {
        ...appState,
        modal: Object.assign(
          appState.modal,
          { open: false },
          { modalState: { name: null, props: {} } },
          { previousModalState: appState.modal.modalState },
        ),
      };

    case actionConstants.CLEAR_ACCOUNT_DETAILS:
      return {
        ...appState,
        accountDetail: {
          privateKey: '',
        },
      };
    case actionConstants.SHOW_SEND_TOKEN_PAGE:
      return {
        ...appState,
        warning: null,
      };

    case actionConstants.LOCK_METAMASK:
      return {
        ...appState,
        warning: null,
      };

    // accounts
    case actionConstants.GO_HOME:
      return {
        ...appState,
        accountDetail: {
          privateKey: '',
        },
        warning: null,
      };

    case actionConstants.SHOW_ACCOUNTS_PAGE:
      return {
        ...appState,
        isLoading: false,
        warning: null,
        scrollToBottom: false,
      };

    case actionConstants.SHOW_CONF_TX_PAGE:
      return {
        ...appState,
        txId: action.id,
        warning: null,
        isLoading: false,
      };

    case actionConstants.COMPLETED_TX:
      return {
        ...appState,
        warning: null,
        txId: null,
      };

    case actionConstants.UNLOCK_FAILED:
      return {
        ...appState,
        warning: action.value || 'Incorrect password. Try again.',
      };

    case actionConstants.UNLOCK_SUCCEEDED:
      return {
        ...appState,
        warning: '',
      };

    case actionConstants.SET_HARDWARE_WALLET_DEFAULT_HD_PATH: {
      const { device, path } = action.payload;
      const newDefaults = { ...appState.defaultHdPaths } as any;
      newDefaults[device] = path;

      return {
        ...appState,
        defaultHdPaths: newDefaults,
      };
    }

    case actionConstants.SHOW_LOADING:
      return {
        ...appState,
        isLoading: true,
        loadingMessage: action.payload,
      };

    case actionConstants.HIDE_LOADING:
      return {
        ...appState,
        isLoading: false,
      };

    case actionConstants.DISPLAY_WARNING:
      return {
        ...appState,
        warning: action.payload,
        isLoading: false,
      };

    case actionConstants.HIDE_WARNING:
      return {
        ...appState,
        warning: undefined,
      };

    case actionConstants.SHOW_PRIVATE_KEY:
      return {
        ...appState,
        accountDetail: {
          privateKey: action.payload,
        },
      };

    case actionConstants.SET_SELECTED_NETWORK_CONFIGURATION_ID:
      return {
        ...appState,
        selectedNetworkConfigurationId: action.payload,
      };

    case actionConstants.SET_NEW_NETWORK_ADDED: {
      const { networkConfigurationId, nickname } = action.payload;
      return {
        ...appState,
        newNetworkAddedName: nickname,
        newNetworkAddedConfigurationId: networkConfigurationId,
      };
    }
    case actionConstants.SET_NEW_TOKENS_IMPORTED:
      return {
        ...appState,
        newTokensImported: action.payload,
      };

    case actionConstants.SET_NEW_TOKENS_IMPORTED_ERROR:
      return {
        ...appState,
        newTokensImportedError: action.payload,
      };

    case actionConstants.SET_NEW_NFT_ADDED_MESSAGE:
      return {
        ...appState,
        newNftAddedMessage: action.payload,
      };

    case actionConstants.SET_REMOVE_NFT_MESSAGE:
      return {
        ...appState,
        removeNftMessage: action.payload,
      };

    case actionConstants.SET_REQUEST_ACCOUNT_TABS:
      return {
        ...appState,
        requestAccountTabs: action.value,
      };

    case actionConstants.SET_OPEN_METAMASK_TAB_IDS:
      return {
        ...appState,
        openMetaMaskTabs: action.payload,
      };

    case actionConstants.HIDE_WHATS_NEW_POPUP:
      return {
        ...appState,
        showWhatsNewPopup: false,
      };

    case actionConstants.CAPTURE_SINGLE_EXCEPTION:
      return {
        ...appState,
        singleExceptions: {
          ...appState.singleExceptions,
          [action.value]: null,
        },
      };

    case actionConstants.TOGGLE_GAS_LOADING_ANIMATION:
      return {
        ...appState,
        gasLoadingAnimationIsShowing: action.payload,
      };

    case actionConstants.SET_WEBHID_CONNECTED_STATUS:
      return {
        ...appState,
        ledgerWebHidConnectedStatus: action.payload,
      };

    case actionConstants.SET_LEDGER_TRANSPORT_STATUS:
      return {
        ...appState,
        ledgerTransportStatus: action.payload,
      };
    case actionConstants.TOGGLE_CURRENCY_INPUT_SWITCH:
      return {
        ...appState,
        sendInputCurrencySwitched: !appState.sendInputCurrencySwitched,
      };
    case actionConstants.ONBOARDED_IN_THIS_UI_SESSION:
      return {
        ...appState,
        onboardedInThisUISession: action.payload,
      };
    case actionConstants.SET_CUSTOM_TOKEN_AMOUNT:
      return {
        ...appState,
        customTokenAmount: action.payload,
      };
    ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
    case actionConstants.SHOW_KEYRING_SNAP_REMOVAL_RESULT:
      return {
        ...appState,
        showKeyringRemovalSnapModal: true,
        keyringRemovalSnapModal: {
          ...action.payload,
        },
      };
    case actionConstants.HIDE_KEYRING_SNAP_REMOVAL_RESULT:
      return {
        ...appState,
        showKeyringRemovalSnapModal: false,
        keyringRemovalSnapModal: {
          snapName: '',
          result: 'none',
        },
      };
    ///: END:ONLY_INCLUDE_IF

    default:
      return appState;
  }
}

// Action Creators
export function hideWhatsNewPopup(): Action {
  return {
    type: actionConstants.HIDE_WHATS_NEW_POPUP,
  };
}

export function toggleGasLoadingAnimation(
  payload: boolean,
): PayloadAction<boolean> {
  return { type: actionConstants.TOGGLE_GAS_LOADING_ANIMATION, payload };
}

export function setLedgerWebHidConnectedStatus(
  payload: WebHIDConnectedStatuses,
): PayloadAction<WebHIDConnectedStatuses> {
  return { type: actionConstants.SET_WEBHID_CONNECTED_STATUS, payload };
}

export function setLedgerTransportStatus(
  payload: HardwareTransportStates,
): PayloadAction<HardwareTransportStates> {
  return { type: actionConstants.SET_LEDGER_TRANSPORT_STATUS, payload };
}

export function toggleCurrencySwitch(): Action {
  return { type: actionConstants.TOGGLE_CURRENCY_INPUT_SWITCH };
}

export function setOnBoardedInThisUISession(
  payload: boolean,
): PayloadAction<boolean> {
  return { type: actionConstants.ONBOARDED_IN_THIS_UI_SESSION, payload };
}

export function setCustomTokenAmount(payload: string): PayloadAction<string> {
  return { type: actionConstants.SET_CUSTOM_TOKEN_AMOUNT, payload };
}

// Selectors
export function getQrCodeData(state: AppSliceState): {
  type?: string | null;
  values?: { address?: string | null };
} | null {
  return state.appState.qrCodeData;
}

export function getGasLoadingAnimationIsShowing(state: AppSliceState): boolean {
  return state.appState.gasLoadingAnimationIsShowing;
}

export function getLedgerWebHidConnectedStatus(
  state: AppSliceState,
): string | null {
  return state.appState.ledgerWebHidConnectedStatus;
}

export function getLedgerTransportStatus(state: AppSliceState): string | null {
  return state.appState.ledgerTransportStatus;
}
