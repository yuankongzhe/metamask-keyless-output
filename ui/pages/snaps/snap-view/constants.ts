export const KeyringSnapRemovalResultStatus = {
  Success: 'success',
  Failed: 'failed',
  None: 'none',
};
