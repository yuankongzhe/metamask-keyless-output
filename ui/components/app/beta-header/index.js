import React from 'react';
import { useI18nContext } from '../../../hooks/useI18nContext';

import {
  TextVariant,
  Color,
  BlockSize,
  Display,
  AlignItems,
  IconColor,
  TextColor,
} from '../../../helpers/constants/design-system';
import { BETA_BUGS_URL } from '../../../helpers/constants/beta';

import { hideBetaHeader } from '../../../store/actions';

import {
  ButtonIcon,
  ButtonIconSize,
  IconName,
  Text,
  Box,
} from '../../component-library';

const BetaHeader = () => {
  const t = useI18nContext();

  return (
    <Box
      display={Display.Flex}
      width={BlockSize.Full}
      backgroundColor={Color.warningDefault}
      padding={2}
      className="beta-header"
      alignItems={AlignItems.center}
    >
      <Text
        variant={TextVariant.bodySm}
        as="h6"
        className="beta-header__message"
        color={TextColor.warningInverse}
      >
        {t('betaHeaderText', [
          <a
            href={BETA_BUGS_URL}
            key="link"
            target="_blank"
            rel="noreferrer noopener"
          >
            {t('here')}
          </a>,
        ])}
      </Text>
      <ButtonIcon
        iconName={IconName.Close}
        size={ButtonIconSize.Sm}
        color={IconColor.warningInverse}
        className="beta-header__button"
        data-testid="beta-header-close"
        onClick={() => {
          hideBetaHeader();
        }}
        aria-label={t('close')}
      />
    </Box>
  );
};

export default BetaHeader;
