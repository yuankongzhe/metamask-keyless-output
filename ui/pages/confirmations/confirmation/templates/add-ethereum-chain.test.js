import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { waitFor } from '@testing-library/react';

import { NetworkStatus } from '@metamask/network-controller';
import { renderWithProvider } from '../../../../../test/lib/render-helpers';
import { MESSAGE_TYPE } from '../../../../../shared/constants/app';

import Confirmation from '../confirmation';

jest.mock('../../../../../shared/lib/fetch-with-cache');

const middleware = [thunk];

const mockApprovalId = 1;
const mockApproval = {
  id: mockApprovalId,
  origin: 'https://test-dapp.metamask.io',
  requestData: {
    rpcUrl: 'https://rpcurl.test.chain',
    rpcPrefs: {
      blockExplorerUrl: 'https://blockexplorer.test.chain',
    },
    chainName: 'Test chain',
    ticker: 'TST',
    chainId: '0x9999',
    nickname: 'Test chain',
  },
};

const mockBaseStore = {
  metamask: {
    pendingApprovals: {
      [mockApprovalId]: mockApproval,
    },
    approvalFlows: [{ id: mockApprovalId, loadingText: null }],
    subjectMetadata: {},
    providerConfig: {
      type: 'rpc',
      rpcUrl: 'http://example-custom-rpc.metamask.io',
      chainId: '0x9999',
      nickname: 'Test initial state',
    },
    selectedNetworkClientId: 'test-network-client-id',
    networksMetadata: {
      'test-network-client-id': {
        EIPS: {},
        status: NetworkStatus.Available,
      },
    },
  },
};

describe('add-ethereum-chain confirmation', () => {
  it('should match snapshot', async () => {
    const testStore = {
      metamask: {
        ...mockBaseStore.metamask,
        pendingApprovals: {
          [mockApprovalId]: {
            ...mockApproval,
            type: MESSAGE_TYPE.ADD_ETHEREUM_CHAIN,
          },
        },
      },
    };
    const store = configureMockStore(middleware)(testStore);
    const { container, getByText } = renderWithProvider(
      <Confirmation />,
      store,
    );
    await waitFor(() => {
      expect(
        getByText('MetaMask does not verify custom networks.'),
      ).toBeInTheDocument();
      expect(container.querySelector('.callout')).toBeDefined();
      expect(container).toMatchSnapshot();
    });
  });
});
