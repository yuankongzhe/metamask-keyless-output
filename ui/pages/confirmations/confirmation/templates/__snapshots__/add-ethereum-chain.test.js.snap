// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-ethereum-chain confirmation should match snapshot 1`] = `
<div>
  <div
    class="confirmation-page"
  >
    <div
      class="confirmation-page__content"
    >
      <div
        class="box box--margin-top-2 box--flex-direction-row box--justify-content-center box--display-flex"
      >
        <div
          class="network-display chip chip--with-left-icon chip--border-color-border-muted chip--background-color-undefined chip--max-content"
          data-testid="network-display"
        >
          <div
            class="chip__left-icon"
          >
            <div
              class="color-indicator color-indicator--filled color-indicator--color-icon-muted color-indicator--size-xs"
              data-testid="color-icon-icon-muted"
            >
              <span
                class="color-indicator__inner-circle"
              />
            </div>
          </div>
          <span
            class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography chip__label typography--h7 typography--weight-normal typography--style-normal typography--color-text-default"
          >
            Test initial state
          </span>
        </div>
      </div>
      <h3
        class="box box--md:margin-4 box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography typography--h3 typography--weight-bold typography--style-normal typography--align-center typography--color-text-default"
      >
        Allow this site to add a network?
      </h3>
      <h6
        class="box box--md:margin-4 box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography typography--h7 typography--weight-normal typography--style-normal typography--align-center typography--color-text-default"
      >
        This will allow this network to be used within MetaMask.
      </h6>
      <h6
        class="box box--margin-top-1 box--margin-bottom-1 box--display-flex box--flex-direction-column box--align-items-center typography typography--h7 typography--weight-normal typography--style-normal typography--color-text-default"
      >
        <b>
          MetaMask does not verify custom networks. 
        </b>
        <span>
           
          Learn about 
          <a
            href="https://metamask.zendesk.com/hc/en-us/articles/4404424659995"
            target="__blank"
          >
            scams and network security risks
          </a>
          .
           
        </span>
      </h6>
      <div
        class="box box--margin-6 box--padding-4 box--padding-bottom-3 box--flex-direction-row box--rounded-lg box--border-color-border-muted box--border-style-solid box--border-width-1"
      >
        <dl
          class="definition-list"
        >
          <dt
            class="box box--margin-bottom-1 box--flex-direction-row typography definition-list__term typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
          >
            Network name
            <div>
              <div
                aria-describedby="tippy-tooltip-1"
                class="definition-list__tooltip-wrapper"
                data-original-title="The name associated with this network."
                data-tooltipped=""
                style="display: inline;"
                tabindex="0"
              >
                <span
                  class="mm-box mm-icon mm-icon--size-sm mm-box--margin-left-1 mm-box--display-inline-block mm-box--color-icon-default"
                  style="mask-image: url('./images/icons/info.svg');"
                />
              </div>
            </div>
          </dt>
          <dd
            class="box box--margin-bottom-2 box--flex-direction-row typography definition-list__definition typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative typography--overflowwrap-break-word"
          >
            Test chain
          </dd>
          <dt
            class="box box--margin-bottom-1 box--flex-direction-row typography definition-list__term typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
          >
            Network URL
            <div>
              <div
                aria-describedby="tippy-tooltip-2"
                class="definition-list__tooltip-wrapper"
                data-original-title="The URL used to access this network."
                data-tooltipped=""
                style="display: inline;"
                tabindex="0"
              >
                <span
                  class="mm-box mm-icon mm-icon--size-sm mm-box--margin-left-1 mm-box--display-inline-block mm-box--color-icon-default"
                  style="mask-image: url('./images/icons/info.svg');"
                />
              </div>
            </div>
          </dt>
          <dd
            class="box box--margin-bottom-2 box--flex-direction-row typography definition-list__definition typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative typography--overflowwrap-break-word"
          >
            https://rpcurl.test.chain
          </dd>
          <dt
            class="box box--margin-bottom-1 box--flex-direction-row typography definition-list__term typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
          >
            Chain ID
            <div>
              <div
                aria-describedby="tippy-tooltip-3"
                class="definition-list__tooltip-wrapper"
                data-original-title="The chain ID used to sign transactions for this network."
                data-tooltipped=""
                style="display: inline;"
                tabindex="0"
              >
                <span
                  class="mm-box mm-icon mm-icon--size-sm mm-box--margin-left-1 mm-box--display-inline-block mm-box--color-icon-default"
                  style="mask-image: url('./images/icons/info.svg');"
                />
              </div>
            </div>
          </dt>
          <dd
            class="box box--margin-bottom-2 box--flex-direction-row typography definition-list__definition typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative typography--overflowwrap-break-word"
          >
            39321
          </dd>
          <dt
            class="box box--margin-bottom-1 box--flex-direction-row typography definition-list__term typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
          >
            Currency symbol
            <div>
              <div
                aria-describedby="tippy-tooltip-4"
                class="definition-list__tooltip-wrapper"
                data-original-title="The ticker symbol displayed for this networkâ€™s currency."
                data-tooltipped=""
                style="display: inline;"
                tabindex="0"
              >
                <span
                  class="mm-box mm-icon mm-icon--size-sm mm-box--margin-left-1 mm-box--display-inline-block mm-box--color-icon-default"
                  style="mask-image: url('./images/icons/info.svg');"
                />
              </div>
            </div>
          </dt>
          <dd
            class="box box--margin-bottom-2 box--flex-direction-row typography definition-list__definition typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative typography--overflowwrap-break-word"
          >
            TST
          </dd>
        </dl>
        <a
          class="button btn-link truncated-definition-list__view-more"
          role="button"
          tabindex="0"
        >
          View all details
        </a>
      </div>
    </div>
    <div
      class="confirmation-footer"
    >
      <div
        class="confirmation-footer__actions"
      >
        <button
          class="button btn--rounded btn-secondary"
          data-testid="confirmation-cancel-button"
        >
          Cancel
        </button>
        <button
          class="button btn--rounded btn-primary"
          data-testid="confirmation-submit-button"
        >
          Approve
        </button>
      </div>
    </div>
  </div>
</div>
`;
